packages/dom/src/focusable.js

```diff
@@ -88,4 +93,20 @@ export function find( context ) {
 
 		return true;
 	} );
+
+	// Firefox treats elements which create a scrollable region/container as
+	// "focusable". Detecting these is expensive, so this is an opt-in.
+	// see https://html.spec.whatwg.org/multipage/interaction.html#focusable-area
+	if ( options.includeScrollable ) {
+		const allChildren = context.querySelectorAll( '*' );
+		const scrollableContainers = [ ...allChildren ].filter( ( el ) => {
+			const hasValidOverflow = window.getComputedStyle( el ).overflow !== 'visible';
+			const hasScroll = el.clientHeight <= el.scrollHeight;
+			return hasValidOverflow && hasScroll;
+		} );
+
+		focusable = [ ...focusable, ...scrollableContainers ];
+	}
```

Comment:
Are we concerned about potential performance impacts here?

Reply:
I'm probably being naive, but is there something about the merging of the two arrays that is specifically concerning? I assume because we could have duplicates in both arrays?

We could use https://lodash.com/docs/4.17.15#uniq to de-dupe to avoid unecessary Nodes being returned.

I'd value any additional guidance here.

Other than this, accessing the DOM layout properties is very expensive in terms of layout reflow/paint...etc. Unfortunately, there isn't a way around this that I'm aware of.

I've added a minor optimisation to avoid the call to check the `overflow` property if there is no scroll detected. This should help.

Reply:
Oh sorry I was not talking about merging arrays but more. The querySelector call and this

> Other than this, accessing the DOM layout properties is very expensive in terms of layout reflow/paint...etc. Unfortunately, there isn't a way around this that I'm aware of.

I understand that there's no way around it but how many times we're calling this function... is maybe where we can guarantee that it's not something recurrent that will harm us.

Reply:
> The querySelector call

I've replaced with this with `getElementsByTagName(*)` because it's a faster performing API.

> ...how many times we're calling this function...